packages/components/organisms/f-registration/src/components/Registration.vue

```diff
@@ -26,107 +26,92 @@
-                            {{ copy.validationMessages.lastName.invalidCharError }}
+                            :class="$style['c-registration-genericError']">
+                            {{ describeLastnameErrorMessage }}
                         </error-message>
                     </template>
                 </form-field>
 
                 <form-field
+                    ref="email"
                     v-model="email"
+                    aria-required="true"
                     name="email"
+                    aria-describedby="error-message-email"
+                    :aria-invalid="!!describeEmailErrorMessage"
                     :label-text="copy.labels.email"
                     input-type="email"
                     @blur="formFieldBlur('email')">
-                    <template #error>
-                        <error-message
-                            v-if="shouldShowEmailRequiredError"
-                            data-test-id='error-email-empty'>
-                            {{ copy.validationMessages.email.requiredError }}
-                        </error-message>
-                        <error-message
-                            v-else-if="shouldShowEmailInvalidError"
-                            data-test-id='error-email-invalid'>
-                            {{ copy.validationMessages.email.invalidEmailError }}
-                        </error-message>
-                        <error-message
-                            v-if="shouldShowEmailMaxLengthError"
-                            data-test-id='error-email-maxlength'>
-                            {{ copy.validationMessages.email.maxLengthError }}
-                        </error-message>
+                    <template
+                        v-if="describeEmailErrorMessage"
+                        #error>
                         <error-message
-                            v-else-if="shouldShowEmailAlreadyExistsError"
-                            data-test-id='error-email-exists'>
-                            {{ copy.validationMessages.email.alreadyExistsError }}
+                            :class="$style['c-registration-genericError']">
+                            {{ describeEmailErrorMessage }}
                         </error-message>
                     </template>
                 </form-field>
 
                 <form-field
+                    ref="password"
                     v-model="password"
+                    aria-required="true"
+                    aria-describedby="error-message-password"
+                    :aria-invalid="!!describePasswordErrorMessage"
                     name="password"
                     :label-text="copy.labels.password"
                     input-type="password"
                     @blur="formFieldBlur('password')">
-                    <template #error>
+                    <template
+                        v-if="describePasswordErrorMessage"
```

Comment:
Probably better to replace a lot of the `v-if` conditions with `v-show`

"use v-if if the condition won't change that often. Use v-show if you want to toggle the condition more often."